### PR TITLE
fix(chip): spacing & slots

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Fixing issues with padding and slots in the Chip component.

**Solving issue**  
Fixes: [DTS-2074](https://tegel.atlassian.net/browse/DTS-2074)

**How to test**  
1. Go to Netlify preview link
2. Check that component has all slots mentioned under Notes tab
3. Check that component padding is correct for all states of component: large, small, without icons, with prefix, with suffix...

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

**Additional context**  
I had to introduce slots for prefixes and suffixes in order to assign correct padding to the label as it changes if icons are present or not. 

[DTS-2074]: https://tegel.atlassian.net/browse/DTS-2074?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alexandereneroth — CHANGES_REQUESTED** [review]
Nice! 🙌

I found 1 issue and 1 potential issue to check before approval:
- Need to fix the the radio/checkbox types in storybook as well
Screenshot 2023-07-12 at 17 19 28

- The star icon is not centered, I wonder if this is a problem with the star icon or the chip component?
👇 
Screenshot 2023-07-12 at 17 09 36

**Comment by theJohnnyMe** [comment]
> Nice! 🙌
> 
> I found 1 issue and 1 potential issue to check before approval:
> 
> * Need to fix the the radio/checkbox types in storybook as well
> 
> Screenshot 2023-07-12 at 17 19 28
> * The star icon is not centered, I wonder if this is a problem with the star icon or the chip component?
>   👇
> 
> Screenshot 2023-07-12 at 17 09 36

Nice finding Alex, I think I just forgot to update story files of Checkbox and Radio button type, on it now ;)

File: core/src/components/chip/chip.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import { Component, Host, h, Event, EventEmitter, Prop } from '@stencil/core';
-import { generateUniqueId } from '../../utils/utils';
+import { Component, Host, h, Event, EventEmitter, Prop, Element } from '@stencil/core';
+import { generateUniqueId, hasSlot } from '../../utils/utils';
 
 /**
+ * @slot prefix - Slot for the prefix icon.
  * @slot label - Slot for the label text.
+ * @slot suffix- Slot for the suffix icon.
```
Comment (nathalielindqvist) [inline core/src/components/chip/chip.tsx:7]:
```suggestion
 * @slot suffix - Slot for the suffix icon.
```

Comment (nathalielindqvist) [reply core/src/components/chip/chip.tsx:7]: Missing space before the dash lead to the slot docs not being rendered properly in the readme

**Review by nathalielindqvist — DISMISSED** [review]
Great work! Just a small comment on the slot docs, otherwise looks good to me 🔥

**Review by nathalielindqvist — APPROVED** [review]

**Review by alexandereneroth — APPROVED** [review]
💯

**Pull request merged**